### ISSUE TITLE
Port tslint-microsoft-contrib's no-increment-decrement

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3556]
See https://github.com/palantir/tslint/issues/1142.

I remember reading a book (JavaScript: The Good Parts? Clean Code? The Art of Unit Testing?) by one of the big influential programming folks and encountering a passage arguing in favor of this type of rule. The justification was something like these points:
* It's easy to type `+i` or `-i` instead of `--i`
* Standardizing widths of increments/decrements makes them easier to read
* It's easy to forget the difference between `++i` and `i++`

Perhaps this rule should be named `increment-decrement` and only ban the pre- operators, to offend fewer people? The first point is the only one of the three I feel would withstand a real debate.

**Comment by ajafff** [comment]
> It's easy to type `+i` or `-i` instead of `--i`

when used as an ExpressionStatement, `no-unused-expression` would warn you about that.
This is only an issue when assigning the result.
But even then you get an error from `prefer-const` because you never reassign the variable (if this is the only place you might have modified it).

**Comment by JoshuaKGoldberg** [comment]
Counter-example:

```typescript
const items = ["apple", "banana", "cherry"];
let index = 0;

while (+index) {
    // ^
    if (items[index][0] === "b") {
        ++index;
    }
}
```

**Comment by JoshuaKGoldberg** [comment]
Fixed by #3575.
